### PR TITLE
Add validation logic

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#11656

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -16,16 +18,40 @@ import (
 )
 
 func main() {
-	config, err := config.Parse()
+	err := mainWithError()
 	if err != nil {
-		panic(microerror.JSON(err))
+		panic(fmt.Sprintf("%#v\n", err))
 	}
+}
```
Comment (rossf7) [inline main.go:25]: I updated the wiring based on the approach in azure-admission-controller.

File: pkg/app/validate_app.go [hunk 2]
```diff
@@ -54,14 +76,15 @@ func (v *Validator) Validate(request *v1beta1.AdmissionRequest) (bool, error) {
...
 		return false, microerror.Mask(err)
 	}
 
-	return appAllowed, nil
-}
+	v.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("admitted app %#q in namespace %#q", app.Name, app.Namespace))
 
-func (v *Validator) ValidateApp(ctx context.Context, app v1alpha1.App) (bool, error) {
-	return true, nil
+	return appAllowed, nil
```
Comment (rossf7) [inline pkg/app/validate_app.go:89]: Adding the validation and some basic logging.

**Review by rossf7 — COMMENTED** [review]
Tested on `giraffe`.

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**